src/test/ui/closures/2229_closure_analysis/nested-closure.rs

```diff
@@ -0,0 +1,48 @@
+// FIXME(arora-aman) add run-pass once 2229 is implemented
+
+#![feature(capture_disjoint_fields)]
+//~^ warning the feature `capture_disjoint_fields` is incomplete
+#![feature(rustc_attrs)]
+
+struct Point {
+    x: i32,
+    y: i32,
+}
+
+// This testcase ensures that nested closures are handles properly
+// - The nested closure is analyzed first.
+// - The capture kind of the nested closure is accounted for by the enclosing closure
+// - Any captured path by the nested closure that starts off a local variable in the enclosing
+// closure is not listed as a capture of the enclosing closure.
+
+fn main() {
+    let mut p = Point { x: 5, y: 20 };
+
+    let mut c1 = #[rustc_capture_analysis]
+        //~^ ERROR: attributes on expressions are experimental
+    || {
+        println!("{}", p.x);
+        //~^ ERROR: Capturing p[(0, 0)] -> ImmBorrow
+        //~^^ ERROR: Min Capture p[(0, 0)] -> ImmBorrow
```

Comment:
Doesn't this outer closure have to capture `p[(1, 0)]`, so that it can be given to the inner closure when that inner closure is created?

Reply:
The enclosing closure captures that, we capture it via MutBorrow which happens in the nested closure. (See the double error thrown for `p[1, 0]` thrown within `c2`.